### ISSUE TITLE
The new modal Safari-style findbar breaks VimFx Find commands (`/` and friends)

### ISSUE THREAD

**Issue by XrXr** [akhodakivskiy/VimFx#760]
The new search mode in FF50 is causing troubles
1. `n` and `N` bring up the search prompt when they shouldn't.
2. Finding a link via `/`, then hitting enter twice would bring up the search prompt instead of following the link

VimFx version: 0.17.1
Firefox version: 50.0a1
OS: Arch Linux
Keyboard language: en-US
Worked before: Firefox 49
Example URL(s): None

**Comment by lydell** [comment]
Yeah, I noticed this as well this morning. Looks like Mozilla is trying out some cool new feature for the find bar. Let's wait and see what stays and what goes before doing anything.

**Comment by lydell** [comment]
@XrXr I just updated my Firefox Developer Edition installation to 50.0a2. It does not seem to contain the new findbar experiments, and the VimFx commands seem to work fine. Is that just me, or is that your experience as well?

**Comment by XrXr** [comment]
That would be the experience of most users, since modal highlight search has been off by default for a while now.
https://bugzilla.mozilla.org/show_bug.cgi?id=1281462
Search is still broken with `findbar.modalHighlight` set to true, but since most people won't have it on maybe it's fine?

**Comment by lydell** [comment]
I see. Haven't really been keeping up-to-date on Firefox development for a while. Not sure it is worth doing anything yet for such an experimental feature, though.

**Comment by XrXr** [comment]
FF 50 is scheduled to release in November. When we get close enough, I imagine the features would be on by default in Nightly if they intend to ship it. We can wait until then.

**Comment by aleksandrs-ledovskis** [comment]
Now it's back with vengeance again - https://bugzilla.mozilla.org/show_bug.cgi?id=1291284

Any poor soul also affected by this change can just set both these preferences to `false` via `about:config`:

```
findbar.highlightAll
findbar.modalHighlight
```

**Comment by lydell** [comment]
Thanks for the heads up and for letting people know how to disable it, @aleksandrs-ledovskis! I think nothing will be done in VimFx until it is on by default in Firefox Developer Edition.

**Comment by SageEx** [comment]
@lydell It is on by default on Nightly and Aurora.
@aleksandrs-ledovskis Thanks for the solution.

**Comment by lydell** [comment]
@SageEx As far as I understand, Aurora has been replaced with Developer Edition. I have Firefox Developer Edition `51.0a2 (2016-10-01) (64-bit)` installed, and it says that no updates are available. I just created a new profile, and the new findbar stuff was _not_ enabled. (It _is_ enabled for me in the latest Nightly, though.) Please enlighten me! :)

**Comment by aleksandrs-ledovskis** [comment]
[Currently](https://bugzilla.mozilla.org/show_bug.cgi?id=1303248) it's behind `#ifdef NIGHTLY_BUILD`& not enabled in downstream version, _unless_ you manually put these preferences to `true`

**Comment by hackel** [comment]
Running FF Developer Edition (51.0a2), and when I press / or even Ctrl-F, the find bar appears, but without focus.  The only way I have found to use it is to actually click into the find bar with the mouse!  Is this a different issue?  I have verified that both findbar.highlightAll and findbar.modalHighlight are set to false.  If I click into the find bar and search for something, I can then use n/N to go to matches.
